plugin/metaprotocol/generator.go

```diff
@@ -35,6 +37,48 @@ func NewGenerator() *Generator {
 
 // Generate create EnvoyFilters for MetaProtocol services
 func (*Generator) Generate(context *model.EnvoyFilterContext) ([]*model.EnvoyFilterWrapper, error) {
+	if context.Gateway != nil {
+		return generateGatewayEnvoyFilters(context)
+	}
+	return generateSidecarEnvoyFilters(context)
+}
+
+func generateGatewayEnvoyFilters(context *model.EnvoyFilterContext) ([]*model.EnvoyFilterWrapper, error) {
+	var envoyfilters []*model.EnvoyFilterWrapper
+	for _, server := range context.Gateway.Spec.Servers {
+		if server.Port == nil {
+			continue
+		}
+		if !protocol.GetLayer7ProtocolFromPortName(server.Port.Name).IsMetaProtocol() {
+			continue
+		}
+		port := trans2Port(server)
+		outboundProxy, err := buildOutboundProxy(context, port)
+		if err != nil {
+			return nil, err
+		}
+		envoyfilters = append(envoyfilters,
+			envoyfilter.GenerateReplaceNetworkFilter(
+				context.ServiceEntry,
+				port,
+				outboundProxy,
+				nil,
+				"envoy.filters.network.meta_protocol_proxy",
+				"type.googleapis.com/aeraki.meta_protocol_proxy.v1alpha.MetaProtocolProxy")...)
+		// append workloadSelector for OutboundListener EnvoyFilter
+		for i := range envoyfilters {
+			envoyfilters[i].Name = fmt.Sprintf("aeraki-router-outbound--%s-%d", context.Gateway.Name, port.Number)
```

Comment:
The current EnvoyFilter name looks like ```aeraki-router-outbound--istio-ingressgateway-909```.  The word router in EnvoyFilter name is too vague. How about  ```aeraki-gateway-$(gateway-name)-$(port)``` ?

Reply:
It's reasonable.